### PR TITLE
MLI extensions for differential privacy

### PR DESCRIPTION
**Pull request by qati**

### CODE REVIEW

File: colearn_grpc/utils.py [hunk 1]
```diff
@@ -91,19 +96,39 @@ async def iterator_to_weights_async(request_iterator, decode=True) -> Weights:
...
-        return Weights(weights=weights_bytes)
+        return Weights(weights=weights_bytes, training_summary=training_summary)
+
+
+def transform_training_summary_to_pb(ts: TrainingSummary) -> TrainingSummaryPb:
+    tsn = TrainingSummaryPb()
+    for field in ["dp_budget"]:
+        target = getattr(tsn, field)
+        source = getattr(ts, field)
+        for entry in source.__fields_set__:
```
Comment (lrahmani) [inline colearn_grpc/utils.py:112]: interesting way of setting protbuf structs, isn't supposed to have object getters and setter for all the fields?

File: colearn_grpc/utils.py [hunk 2]
```diff
@@ -91,19 +96,39 @@ async def iterator_to_weights_async(request_iterator, decode=True) -> Weights:
...
+            setattr(target, entry, getattr(source, entry))
+    return tsn
+
+
+def transform_training_summary_from_pb(ts: TrainingSummaryPb) -> TrainingSummary:
+    tsn = TrainingSummary()
+    for field in ["dp_budget"]:
+        target = getattr(tsn, field)
+        source = getattr(ts, field)
+        for entry in target.__fields__.keys():
```
Comment (lrahmani) [inline colearn_grpc/utils.py:122]: same as above

**Review by lrahmani — APPROVED** [review]
LGTM

File: colearn_grpc/utils.py (hunk 1 above)
Comment (qati) [reply colearn_grpc/utils.py:112]: they have all kinds of helpers, but this is the easiest way of getting the fields.

**Review by evsmithx — APPROVED** [review]

**Pull request merged**